core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -684,17 +689,16 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
   }
 
   private def checkpointBrokerId(brokerId: Int) {
```

Comment:
This code is only called at startup time. It is to protect against data messed up from different brokers. Since the broker hasn't started yet, It is probably OK for the broker to shutdown. Users can remove the log dir from the path and restart the broker in this case.